<launch>
  <arg name="control_performance_analysis_param_path" default="$(find-pkg-share autoware_control_performance_analysis)/config/control_performance_analysis.param.yaml"/>
  <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="input/control_raw" default="/control/command/control_cmd"/>
  <arg name="input/measured_steering" default="/vehicle/status/steering_status"/>
  <arg name="input/current_odometry" default="/localization/kinematic_state"/>
  <arg name="output/error_stamped" default="/control_performance/performance_vars"/>
  <arg name="output/driving_status_stamped" default="/control_performance/driving_status"/>

  <!-- vehicle info -->
  <arg name="vehicle_info_param_file" default="$(find-pkg-share autoware_vehicle_info_utils)/config/vehicle_info.param.yaml"/>

  <node pkg="autoware_control_performance_analysis" exec="autoware_control_performance_analysis_node" name="control_performance_analysis" output="screen">
    <param from="$(var control_performance_analysis_param_path)"/>
    <param from="$(var vehicle_info_param_file)"/>

    <remap from="~/input/reference_trajectory" to="$(var input/reference_trajectory)"/>
    <remap from="~/input/control_raw" to="$(var input/control_raw)"/>
    <remap from="~/input/measured_steering" to="$(var input/measured_steering)"/>
    <remap from="~/input/odometry" to="$(var input/current_odometry)"/>
    <remap from="~/output/error_stamped" to="$(var output/error_stamped)"/>
    <remap from="~/output/driving_status_stamped" to="$(var output/driving_status_stamped)"/>
  </node>
</launch>
